test.bats

```diff
@@ -670,3 +670,19 @@ EOF
   [ ! -z $status ]
   [ "$(echo "$output" | jq .)" == "$(echo "$expected" | jq .)" ]
 }
+
+@test "test parseImageName using image starting with underscore" {
+  IMAGE="_something:tag123"
+  TAGVAR=false
+  run parseImageName
+  [ $status -eq 13 ]
+}
+
+@test "test parseImageName using image containing an underscore in tag name" {
+  IMAGE="something:tag_123"
+  TAGVAR=false
+  run parseImageName
+  [ ! -z $status ]
+  [ "$output" == "something:tag_123" ]
+  echo "output = $output" 1>&2
```

Comment:
These three lines of code are arcane shell scripting.

Reply:
Should I care? Do you have a suggestion?

Reply:
For those reading this after the fact, @mtompset and I discussed this directly. We determined that since this is using a test framework (bats) that may affect how this is implemented, the arcane scripting can be left as it is.